src/libmongoc/src/mongoc/mongoc-util.c

```diff
@@ -688,6 +688,29 @@ _mongoc_getenv (const char *name)
 #endif
 }
 
+bool
+_mongoc_setenv (const char *name, const char *value)
+{
+#ifdef _WIN32
+   char *envstring;
+
+   envstring = bson_strdup_printf ("%s=%s", name, value);
```

Comment:
Missing a free of this string (and it always has?)

---

Suggest updating this to use [`SetEnvironmentVariableA()`](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setenvironmentvariable) instead (With the `A` suffix since we don't yet care about Unicode support here).